src/StripeTests/Infrastructure/ParameterBuilderTest.cs

```diff
@@ -45,49 +45,293 @@ public void EncodesParameters()
         {
             Assert.NotNull(this.service);
 
-            var testObject = new TestOptions
+            var testCases = new[]
             {
-                StringContainingText = "Foo",
-                StringWithDifferentName = string.Empty,
-                Number = 42,
-                Metadata = new Dictionary<string, string>
+                // No data
+                new
                 {
-                    { "A", "Value-A" },
-                    { "B", "Value-B" },
-                    { "empty_value", string.Empty },
+                    data = new TestOptions { },
+                    want = "?bool=False&decimal=0&enum=test_one&int=0"
```

Comment:
Non-nullable value types will always be sent, since they're initialized with their zero value and we have no way of detecting whether the value was explicitly set by the user or not.

The right way to fix this is probably to force all value types to be nullable (sort of equivalent to using pointers for all attributes in stripe-go).

Reply:
Gotcha, so it's the opposite of go I guess. We'll make everything optional in a following PR.
I think we can merge